### PR TITLE
Handle mixed item origins in multiplayer

### PR DESCRIPTION
**Pull request by AJenbo**
This is the last piece of the puzzle that is 1.2.0

### CODE REVIEW

**Comment by qndel** [comment]
wouldn't it be good to have defines for last valid spell/affix at this point? or just check that in a different way than using a certain hardcoded spell/affix :D

**Comment by AJenbo** [comment]
I did apply the `spell_id`, the rest I don't think has enums, but once they do it should be easy to add them here. Or where are you thinking of ones that are explicitly for the limits? Or were you thinking in the style of `IDI_FIRSTQUEST`? I guess that is an option as well.

File: Source/loadsave.cpp [hunk 1]
```diff
@@ -183,6 +183,23 @@ class SaveHelper {
...
+void RemoveInvalidItem(ItemStruct *pItem)
+{
+	bool isInvalid = !IsItemAvailable(pItem->IDidx) || !IsUniqueAvailable(pItem->_iUid);
+
+	if (!gbIsHellfire) {
+		isInvalid = isInvalid || (pItem->_itype == ITYPE_STAFF && GetSpellStaffLevel(pItem->_iSpell) == -1);
+		isInvalid = isInvalid || (pItem->_iMiscId == IMISC_BOOK && GetSpellBookLevel(pItem->_iSpell) == -1);
+		isInvalid = isInvalid || pItem->_iDamAcFlags != 0;
+		isInvalid = isInvalid || pItem->_iPrePower > IPL_AC_CURSE;
+		isInvalid = isInvalid || pItem->_iSufPower > IPL_AC_CURSE;
```
Comment (qndel) [inline Source/loadsave.cpp:195]: I'm guessing IPL_AC_CURSE is the last valid affix? That's where I was thinking about some stuff other than hardcoded one

File: Source/spells.cpp [hunk 2]
```diff
@@ -361,20 +363,26 @@ int GetSpellBookLevel(int s)
 		case SPL_NOVA:
 		case SPL_APOCA:
 			return -1;
+		default:
+			if (s > SPL_BONESPIRIT)
```
Comment (qndel) [inline Source/spells.cpp:367]: same here, some max spell index define/whatever? :P

Comment (AJenbo) [reply Source/spells.cpp:367]: Can add `SPL_LASTDIABLO` instead.

**Pull request merged**